Userland/Libraries/LibGfx/Font/Emoji.cpp

```diff
@@ -58,18 +58,46 @@ Bitmap const* Emoji::emoji_for_code_point_iterator(Utf8CodePointIterator& it)
     struct EmojiAndCodePoints {
         Bitmap const* emoji;
         Span<u32> code_points;
+        u8 real_codepoint_length;
     };
     Vector<EmojiAndCodePoints, max_emoji_code_point_sequence_length> possible_emojis;
 
     // Determine all existing emojis for the longest possible ZWJ emoji sequence,
     // or until we run out of code points in the iterator.
-    for (size_t i = 0; i < max_emoji_code_point_sequence_length; ++i) {
+    bool last_codepoint_sequence_found = false;
+    for (u8 i = 0; i < max_emoji_code_point_sequence_length; ++i) {
         auto code_point = it.peek(i);
         if (!code_point.has_value())
             break;
-        code_points.append(*code_point);
-        if (auto const* emoji = emoji_for_code_points(code_points))
-            possible_emojis.empend(emoji, code_points);
+        // From https://unicode.org/reports/tr51/#Emoji_Implementation_Notes:
+        // > minimally-qualified or unqualified emoji zwj sequences may be handled
+        // > in the same way as their fully-qualified forms; the choice is up to
+        // > the implementation.
+        //
+        // Serenity chooses to treat minimally-qualified and unqualified
+        // emojis the same as fully-qualified emojis.
+        //
+        // So, if the emoji presentation specifier is found, just skip it in order
+        // to drop fully-qualified emojis down to their minimally-qualified or
+        // unqualified forms. This ensures that all forms are treated the same
+        if (code_point.value() == 0xFE0F) {
+            // If the last sequence was found, then we can just update
+            // its real length.
+            if (last_codepoint_sequence_found) {
+                possible_emojis.last().real_codepoint_length++;
+            }
+            // And we can always skip the lookup since the code point sequence
+            // will be unchanged since last time.
+            continue;
+        } else {
+            code_points.append(*code_point);
+        }
+        if (auto const* emoji = emoji_for_code_points(code_points)) {
+            possible_emojis.empend(emoji, code_points, (u8)(i + 1));
```

Comment:
`i` is already an u8, why would we need a cast?

Reply:
auto-promotion to int I guess; :cplusplus:

Reply:
Yep, was being promoted. Was able to get around this by storing i + 1 in a variable first.